Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -0,0 +1,228 @@
+/-
+Copyright (c) 2023 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import Mathlib.MeasureTheory.Measure.ProbabilityMeasure
+
+/-!
+# The Lévy-Prokhorov distance on spaces of finite measures and probability measures
+
+## Main definitions
+
+* `MeasureTheory.levyProkhorovEDist`: The Lévy-Prokhorov distance between two measures.
```

Comment:
Do you think it matters for the name that it's not real-valued? I would just name it `levyProkhorovDist` for simplicity.

Reply:
It is the `edist` in the corresponding `PseudoEMetricSpace`. For finite measures it is finite, so we can define a `PseudoMetricSpace` (not done in this PR yet).

Reply:
BTW, should we do this?
```
def LevyProkhorov (α : Type*) := α

instance : PseudoEMetricSpace (LevyProkhorov (Measure α)) := _
instance : PseudoMetricSpace (LevyProkhorov (FiniteMeasure α)) := _
instance : PseudoMetricSpace (LevyProkhorov (ProbabilityMeasure α)) := _
```

Reply:
Indeed the idea was that on `FiniteMeasure X` and `ProbabilityMeasure X` the Lévy Prokhorov distance is an actual pseudometric. (And in Borel spaces furthermore an actual metric.) The pseudometric requires an extra post-composition with `ENNReal.toReal`, so I avoided calling this definition just "dist".

I did plan to add a type synonym indeed to do comparisons of the topology with weak convergence, like @urkud suggested. Since the comparisons of the topology are in later PRs (some of which I haven't even written locally), I also did not yet introduce the type synonym in this PR.